### PR TITLE
More hostile species for expeditions

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
### New species to expedition faction pool:
- **Aberrant flesh**. Features sprites by wax391 (discord). Expedition specific version don't breathe and are not affected by low pressure and shouldn't be affected by dangerous/volatile atmos.
- **Argocytes**. Expedition specific version has lower health and damage than base game one (those were kinda bs imo), don't breath, shouldn't be affected by dangerous/volatile atmos, can path through airlocks.
- **Dinosaurs**. Taken from https://github.com/space-wizards/space-station-14/pull/26883 by @Vonsant (I edited ymls though a bit), sprites by belay5 (discord). Changes to original yml were made to allow dinosaurs to survive on planets: they don't breath, shouldn't be affected by dangerous/volatile atmos, can path through airlocks.
- **Mercenaries**. Based on this [idea thread](https://discord.com/channels/1123826877245694004/1243398315937366016/1243398315937366016). Armed with weapons one can obtain from LiberationStation. The mob visuals derived not from gear loadouts but from sprite layering in yml. If this proves to be a viable solution, in time, I'll rework syndicate/cultist/wizards mob in this fashion to cut down entity spam from those mobs.
- **Explorers**. Based on this [idea thread](https://discord.com/channels/1123826877245694004/1243507351982702633/1243507351982702633). Armed with mining tools and gear. The mob visuals derived is mix of loadouts and sprite layering in yml. Supposed to be a beginner friendly enemy, like aberrant flesh or carps.
- **Rogue AI**. Challenging enemy: rogue machines can not be killed - only temporary incapacitated (it takes about 2-3 minutes for robot to raise from dead; word of advice keep that diagnostic hud ready), unlike other expedition enemies ranged rogue borgs deal heat damage. Reliable source of machine parts and circuit boards. Sprites for bots were taken from BayStation12 and Goonstation (both are under CC-BY-SA-3.0 afaict, but Goonstation sprite are also NC). Features sprite randomization.

### Other major changes:
- Made expedition specific versions of carps: they don't breath, shouldn't be affected by dangerous/volatile atmos, can path through airlocks and mob range includes sharkminnows.
- Made expedition specific versions of xenos: they don't breath, shouldn't be affected by dangerous/volatile atmos, can path through airlocks and mob range includes runners.
- Lowered the cost for Xeno `salvageFaction` to increase the chance for encountering them on minimal difficulty.
- Removed artificial reflection from expedition mobs.
- All expedition mobs now have all access and ability to pry open doors.
- All expedition mobs and bluespace event mobs (syndicate agents, blood cultists and wizards) got `MovementIgnoreGravity`.
- Most of the gear from hostile human NPCs can't be stripped off (in reasonable time).
- Mobs that have ranged attack can now shoot over railings.

### Bundled minor changes:
- Fixed solution transfer from syndie darts.
- Cleaned up NPC gloves: removed .rsi files - everything can be done through .yml.
- **Removed** `Gloves.rar` _someone_ left in game files.

### New items:
- IFF strobe: wearable tiny "strobe" - 1 to 3 blinking unshaded pixels. Can be printed from salvage techfab.
- Turbo laser - 4 retro laser pistols in a single package. Requires 2 hands to wield, slows you down by 30%, has a bit of a spread, fun to shoot. 10% chance to loot one from boss enemies on rogue AI expeditions.
- Salvage radar - destruction objective for expeditions with explorers enemy faction.
- Rogue AI Node - destruction objective for expeditions with rogue silicons enemy faction.

## Why / Balance
Some enemy variety for expeditions on low/mid difficulties.

## How to test
Run expeditions or spawn new mobs.

## Media
image
image
image
image
image
image
image

image
image
image
image
- [x] I have added screenshots/videos to this PR showcasing its changes ingame

## Breaking changes
- edits to `\Resources\Prototypes\Procedural\salvage_factions.yml`
- edits to `\Resources\Prototypes\ai_factions.yml`
- moved `\Resources\Prototypes\_NF\Entities\Mobs\NPCs\argocyte.yml` to `\Resources\Prototypes\_NF\Entities\Markers\Spawners\Random\argocyte_mobs.yml` because it's spawner.

**Changelog**
:cl: erhardsteinhauer
- add: Added hostile mercenaries to expedition enemy faction pool.
- add: Added hostile salvagers to expedition enemy faction pool.
- add: Added hostile robots to expedition enemy faction pool.
- add: Added aberrant flesh to expedition enemy faction pool, new sprites by wax391 (discord).
- add: Added argocytes to expedition enemy faction pool.
- add: Added dinosaurs to expedition enemy faction pool. Original mobs by Vonsant, sprites by belay5 (discord).
- add: Wearable IFF strobe. Can be printed from salvage techfab.
- add: Turbo laser.
- fix: Fixed solution transfer from syndie darts.

### CODE REVIEW

**Comment by ErhardSteinhauer** [comment]
Come to think of it, going to wait until this PR is merged: https://github.com/new-frontiers-14/frontier-station-14/pull/1333

**Comment by Vonsant** [comment]
May i add my closed PR from Wizard with Dinosaurs?
https://github.com/space-wizards/space-station-14/pull/26883
I think they will be good at expeditions. Maybe.

**Comment by ErhardSteinhauer** [comment]
> May i add my closed PR from Wizard with Dinosaurs? [space-wizards/space-station-14#26883](https://github.com/space-wizards/space-station-14/pull/26883) I think they will be good at expeditions. Maybe.

I like the idea of dinosaurs in expeds as alternative to xenos for example, but I'm not diggin' those sprites. I'm sure someone of volunteer spriters might be on board with redrawing them. But regardless I'll start preppin' ymls for dinos.

**Comment by dvir001** [comment]
> May i add my closed PR from Wizard with Dinosaurs? [space-wizards/space-station-14#26883](https://github.com/space-wizards/space-station-14/pull/26883) I think they will be good at expeditions. Maybe.

Please do, we will love to have them, but I agree with Erhard that they might need a bit of style edit.

Either way we can add them and edit later if need be.

**Comment by ErhardSteinhauer** [comment]
> Either way we can add them and edit later if need be.

True that. Guess I'll grab sprites then.

**Comment by Vonsant** [comment]
> True that. Guess I'll grab sprites then.

Thanks a lot. What should i do right now? Redo my PR here, or you just take a sprite by youreself?
(My english not so good).

**Comment by ErhardSteinhauer** [comment]
> > True that. Guess I'll grab sprites then.
> 
> Thanks a lot. What should i do right now? Redo my PR here, or you just take a sprite by youreself? (My english not so good).

I already too 'em. Gonna add to this PR. Naturally credits for dinos will go to you (although I'll change ymls a bit because I have general prototype for mobs).

**Comment by ErhardSteinhauer** [comment]
I believe this PR is ready for review

**Comment by ErhardSteinhauer** [comment]
The PR I was waiting was merged, so I guess this PR shouldn't have `Status: Awaiting Changes`. Unless I'm missing something.

**Comment by ErhardSteinhauer** [comment]
ikr

**Comment by dvir001** [comment]
Ok ok, ill look into this

**Review by whatston3 — DISMISSED** [review]
Testing seemed alright.  New enemies killed me often, but I'm not the most robust fighter.  A few small suggestions, but seems fine overall.

I'm assuming the increased cash reward for most explorer and merc mobs is to balance against the syndicate item rewards.

I like the variety, this is a good feature.  I'd like to see an expedition with the explorers in the salvage outpost from your other PR.  That leviathing and those dinosaurs are very tanky, and the first time seeing a boss explorer with the quick shooting PKA is spooky.  I like the IFF strobe, hoping to see salvage workers with a set on sometime.

File: Resources/Locale/en-US/_NF/advertisements/mobchatter/mercenaryhumanoidmob.ftl
Comment (ErhardSteinhauer) [reply Resources/Locale/en-US/_NF/advertisements/mobchatter/mercenaryhumanoidmob.ftl]: mercanary.. oops. fixed.

File: Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/mobs_hostile_argocyte.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/mobs_hostile_argocyte.yml]: Fair, fixed

File: Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/mobs_hostile_dinosaurs.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/mobs_hostile_dinosaurs.yml]: Fair, fixed

File: Resources/Prototypes/Procedural/salvage_factions.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/Procedural/salvage_factions.yml]: Huh, don't remember changing that line to xeno. Fixed.

**Comment by ErhardSteinhauer** [comment]
> Testing seemed alright. New enemies killed me often, but I'm not the most robust fighter. A few small suggestions, but seems fine overall.
> 
> I'm assuming the increased cash reward for most explorer and merc mobs is to balance against the syndicate item rewards.

Their loot isn't stellar, but now that you mentioned it, I  might've been too generous with cash. Probably should tune it down a bit.

> I like the variety, this is a good feature. I'd like to see an expedition with the explorers in the salvage outpost from your other PR. That leviathing and those dinosaurs are very tanky, and the first time seeing a boss explorer with the quick shooting PKA is spooky.

I hope I managed to create a sort of difficulty ladder:
Aberrant flesh ~ Carps < Xeno < Dinosaurs < Explorers < Argocytes < Mercenaries < Cultists < Syndicate

Might need to nerf Argocytes even more in the future (already halved their HP and damage in comparison with the original mobs from the base game).

> I like the IFF strobe

Thanks!

Off topic: running with 2 sawn-off PKAs is stupidly fun. Highly inefficient, can't hit anything reliably, but fun.

**Comment by ErhardSteinhauer** [comment]
Added one more faction - Rogue AIs. Far from polished, but functional and kinda fun from testing.

Probably should draw the line here - no more adding new stuff to this PR - only fixes, otherwise there will be no end to fiddling.

**Comment by Vonsant** [comment]
_So hyped_

**Review by dvir001 — APPROVED** [review]
Cannot find any major issues with this 200+ files PR
After small fixups

**Pull request merged**
